src/libecalc/common/utils/rates.py

```diff
@@ -943,8 +943,15 @@ def merge(self, other: TimeSeries) -> TimeSeriesRate:
         )
 
     def for_period(self, period: Period) -> Self:
+        if period.end < self.timesteps[0] or period.start > self.timesteps[-1]:
+            return self.__class__(
```

Comment:
I like that you create a "null class" instead of None, to make it possible to use it, as it is still valid - albeit empty...